app/src/androidTest/java/org/mozilla/reference/browser/ui/robots/ExternalAppsRobot.kt

```diff
@@ -44,12 +44,21 @@ private fun assertAutofillServices() {
         .waitForExists(waitingTime))
 }
 
+@Suppress("SwallowedException")
 private fun assertYouTubeApp() {
-    mDevice.waitForIdle()
-    assertTrue(mDevice.findObject(UiSelector().text("Home"))
-        .waitForExists(waitingTime))
-    assertTrue(mDevice.findObject(UiSelector().text("Subscriptions"))
-        .waitForExists(waitingTime))
+    try {
+        // Check youtube's home buttons
+        mDevice.waitForIdle()
+        assertTrue(mDevice.findObject(UiSelector().text("Home"))
+            .waitForExists(waitingTime))
+        assertTrue(mDevice.findObject(UiSelector().text("Subscriptions"))
+            .waitForExists(waitingTime))
+    } catch (e: AssertionError) {
+        println("The native youtube app opens but needs to be updated")
+        // In case the app isn't up to date on the emulator an update message will be displayed
+        assertTrue(mDevice.findObject(UiSelector().text("Update for a faster, better YouTube"))
```

Comment:
Can't we just hit back on dialog? This feels really hacky looking for text in another app which could change any point.

Reply:
Tapping  back will just close the 3rd party app (without verifying that indeed the link was properly opened)
I guess it's more or less  the same as when we perform queries using the hard coded text of a button, menu etc...